ner_v2/detectors/temporal/date/standard_regex_date.py

```diff
@@ -0,0 +1,527 @@
+from chatbot_ner.config import ner_logger
+from constants.detection_constant import TYPE_EXACT
+from ner_v2.detectors.temporal.constant import DATE_CONSTANT_FILE, DATETIME_CONSTANT_FILE, \
+    RELATIVE_DATE, DATE_LITERAL_TYPE, MONTH_LITERAL_TYPE, WEEKDAY_TYPE, \
+    MONTH_TYPE, ADD_DIFF_DATETIME_TYPE, MONTH_DATE_REF_TYPE, NUMERALS_CONSTANT_FILE
+import pytz
+import re
+import datetime
+from dateutil.relativedelta import relativedelta
+
+from ner_v2.detectors.temporal.utils import next_weekday, nth_weekday, get_tuple_dict
+
+
+class BaseRegexDate(object):
+    def __init__(self, data_directory_path, timezone='UTC', is_past_referenced=False):
+        """
+        Base Regex class which will be imported by language date class by giving their data folder path
+        This will create standard regex and their parser to detect date for given language.
+        Args:
+            data_directory_path (str): path of data folder for given language
+            timezone (str): user timezone default UTC
+            is_past_referenced (boolean): if the date reference is in past, this is helpful for text like 'kal',
+                                          'parso' to know if the reference is past or future.
+        """
+        self.text = ''
+        self.processed_text = ''
+        self.tag = ''
+
+        try:
+            self.timezone = pytz.timezone(timezone)
+        except Exception as e:
+            ner_logger.debug('Timezone error: %s ' % e)
+            self.timezone = pytz.timezone('UTC')
+            ner_logger.debug('Default timezone passed as "UTC"')
+
+        self.now_date = datetime.datetime.now(tz=self.timezone)
+        self.is_past_referenced = is_past_referenced
+
+        # dict to store words for date, numerals and words which comes in reference to some date
+        self.date_constant_dict = {}
+        self.datetime_constant_dict = {}
+        self.numerals_constant_dict = {}
+
+        # define dynamic created standard regex from language data files
+        self.regex_relative_date = None
+        self.regex_day_diff = None
+        self.regex_date_month = None
+        self.regex_date_ref_month_1 = None
+        self.regex_date_ref_month_2 = None
+        self.regex_date_ref_month_3 = None
+        self.regex_after_days_ref = None
+        self.regex_weekday_month_1 = None
+        self.regex_weekday_month_2 = None
+        self.regex_weekday_diff = None
+        self.regex_weekday = None
+
+        # Method to initialise value in regex
+        self.init_regex_and_parser(data_directory_path)
+
+        # Variable to define default order in which these regex will work
+        self.detector_preferences = [self._detect_relative_date,
```

Comment:
1) Enforce everything inheriting base detector to have detect_date method.
2) Inside Readme.md, add an example of adding a new pattern in a given language. This will include overwriting priorities and overwriting detect_date which you will add here.